src/arch/svm/SpokeUtils.ts

```diff
@@ -97,11 +101,19 @@ export async function getTimestampForSlot(provider: SVMProvider, slotNumber: big
     }
 
     const { __code: code } = err.context;
-    if ([SVM_NO_BLOCK_AT_SLOT].includes(code)) {
-      return undefined;
+    switch (err.context.__code) {
+      case SVM_NO_BLOCK_AT_SLOT:
+        return undefined;
+
+      case SVM_BLOCK_NOT_AVAILABLE:
+        if (retries <= 0) {
+          throw new Error(`Timeout on SVM getBlockTime() for slot ${slotNumber.toString()}`);
+        }
+        return getTimestampForSlot(provider, slotNumber, --retries);
+
+      default:
+        throw new Error(`Unhandled error on SVM getBlockTime() for slot ${slotNumber.toString()}: ${code}`);
```

Comment:
We should still include the full error object here right?

Reply:
Good call - the `cause` field is designed for this: https://github.com/across-protocol/sdk/pull/1154/commits/4ca82ca03539ae961888c75da0841d69e947c601